src/rust/engine/src/python_parsing.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
I'm probably going to move this code into the `dep_inference` crate. Then I can test it as well.